notebooks/build_index_alternative_cynkra.R

```diff
@@ -0,0 +1,123 @@
+#--- note ----------------------------------------------------------------------
+
+# England ltla24 index for power outage vulnerability
+
+# Version of building the index using non-negative matrix factorization.
+# I think it makes sense for this data, where all indicators are higher = worse.
+
+
+#--- init ----------------------------------------------------------------------
+
+library(NMF)
+library(here)
+
+indir <- here("data", "clean", "england", "ltla")
+
+
+#--- load data -----------------------------------------------------------------
+
+files <- list.files(indir, pattern = "eng_.*.csv", full.names = TRUE)
+dat   <- Map(read.csv, files)
+
+names(dat) <- basename(names(dat))
+
+
+#--- clean ---------------------------------------------------------------------
+
+# removing "asylym_support_ltla" since there is "asylum_support_ltla24"
+dat <- dat[names(dat) != "eng_asylum_support_ltla.csv"]
+
+
+#--- merge ---------------------------------------------------------------------
+
+# only select ltla24 codes and z score data from each dataset
+dat <- Map(function(x) x[,grepl("ltla24_|z_", colnames(x))], dat)
+
+# merge all the datasets by ltla24_code column
+dat <- Reduce(function(x,y) merge(x, y, by = "ltla24_code", all.x = TRUE), dat)
+
+# z_asylum_support_perc had fewer rows imputing the missing values with 0
+# note that for z-scores this is equivalent with mean imputation
+dat[is.na(dat)] <- 0
+
+
+#--- rescale -------------------------------------------------------------------
+
+# first - rescale to z-scores just in case
+dat[-1] <- apply(dat[-1], 2, function(x) (x - mean(x)) / sd(x))
+
+# make all data positive, ranging from 0 to 2, with the mean of 1
+# we make the extreme value be either -1 or 1 (depending on the sign)
```

Comment:
Is this comment a mistake? Does this not contradict the statement above about making the data range from 0-2 (to remove negative values)?